mlrun/common/schemas/model_endpoints.py

```diff
@@ -242,6 +242,12 @@ def flat_dict(self):
                     mlrun.common.model_monitoring.EventLiveStats.PREDICTIONS_PER_SECOND: 0,
                 }
             }
+
+        # Validate tht the error count value is an integer - important for the grafana dashboards
```

Comment:
```suggestion
        # Validate that the error count value is an integer - important for the grafana dashboards
```
Is this meant to validate (fail on wrong value) or type cast (a string or float) or both? How do we end up with something other than an int there? Can we add a regression test for it?

Reply:
agree with Gal here, we first need to get to the bottom of how come zero value became a string..

Reply:
If you look at this part of the code you'll see that we are generating a flattened dictionary and therefore applying `json.dumps()` on each non-string and non-boolean values.  The error count value is the only value which is important for us to keep as an integer because of a future calculations. Therefore we convert it "back" to integer. We can't exclude all integers from the `json.dumps()` operation because we also have an enum object with integer values that should be converted to json as well (and python recognizes this enum value as integer).
 
```
# Initialize a flattened dictionary that will be filled with the model endpoint dictionary attributes
flatten_dict = {}
for k_object in model_endpoint_dictionary:
    for key in model_endpoint_dictionary[k_object]:
        # If the value is not from type str or bool (e.g. dict), convert it into a JSON string
        # for matching the database required format
        if not isinstance(
            model_endpoint_dictionary[k_object][key], (str, bool)
        ):
            flatten_dict[key] = json.dumps(
                model_endpoint_dictionary[k_object][key]
            )
        else:
            flatten_dict[key] = model_endpoint_dictionary[k_object][key]
```

From 1.3.1, as part of the redesigning of the model endpoint schema operations, we have started to store this zero value in the KV table as a string (which is a bug) and from this PR it will be stored as an integer as expected.

Reply:
so if I understand correctly we can either:
1. convert all integers to str with json.dumps(), and then "patch" ERROR_COUNT to int
2. exclude all integers from json.dumps(), and then "patch" the enum object to str with json.dumps()

do you see preference to (1) over (2)? 

also it's still not clear to me why current code behaves differently (using str or int) when the error_count == 0 vs. error_count >0, can you explain?

Reply:
Maybe I'm misunderstanding the context, and I have no prior experience with Python enumerations, but I see that
```python
>>> Color.RED
<Color.RED: 1>
>>> type(Color.RED)
<enum 'Color'>
>>> isinstance(Color.RED, int)
False
```

Reply:
@gtopper 
I see now that the enum class that hasn't been converted is actually `enum.IntEnum`. In that case, you'll get:
![image](https://github.com/mlrun/mlrun/assets/101177563/7c9cb364-50f1-4f0a-9653-7b4813cee5d9)

I'll try to change it to a basic `enum` object and then maybe we will be able to exclude integers as well (and therefore to avoid this "reconverting" of the error count into int)

Reply:
> so if I understand correctly we can either:
> 
> 1. convert all integers to str with json.dumps(), and then "patch" ERROR_COUNT to int
> 2. exclude all integers from json.dumps(), and then "patch" the enum object to str with json.dumps()
> 
> do you see preference to (1) over (2)?
> 
> also it's still not clear to me why current code behaves differently (using str or int) when the error_count == 0 vs. error_count >0, can you explain?

@assaf758 
Right now I'm trying to implement (2). The issue with `enum` objects was because one of the objects was `IntEnum` and not `enum` and therefore python considered the value as an int. 

Regarding to your 2nd question I'll try to explain the process and the bug (quite detailed...):
For each model endpoint that the user is deploying, we set by default `error_count=0`. In the current code, this value is being stored as a string. The KV schema is being generated in the monitoring stream graph in a step called `InferSchema` which is executed only if the schema hasn't been generated yet. 
However, one step before that (`UpdateEndpoint`), we update the values of the model endpoint record in the KV table according to the last batch of events, including the `error_count` value. Now, if there have been errors in that last batch of events  (such as in the case of the QA test - https://jira.iguazeng.com/browse/ML-3799 ), then the monitoring stream graph updates the KV record with the amount of errors as an integer value. The problem here is what happen if one of the KV records has no errors  while the other records do have errors (again, such in the QA test)? in that case, this record has `error_count=0` as a string while other records have `error_count=x` as an int. 
The actual bug here is that we stored the `error_count=0` as a string from the 1st place (which also caused issues in grafana that tries to sum the error count values).

Reply:
You also have the alternative of explicitly checking `not isinstance(x, IntEnum)` if for some reason you need to keep `IntEnum`.

Reply:
Thanks, I've updated the PR according to your suggestion :)